cpp/src/stream_compaction/drop_duplicates.cu

```diff
@@ -187,4 +187,16 @@ rows in input table should be equal to number of rows in key colums table");
   nvcategory_gather_table(input_table, destination_table);
   return destination_table;
 }
+
+gdf_size_type unique_count(gdf_column const& input_column)
+{
+  if (0 == input_column.size) {
+    return 0;
+  }
+  gdf_size_type unique_count; 
+  std::tie(std::ignore, unique_count) =
+    detail::get_unique_ordered_indices({const_cast<gdf_column*>(&input_column)}, duplicate_keep_option::KEEP_FIRST, true);
```

Comment:
This is fine for now, but I think you should file an issue to optimize this. If the unique indices are not needed, then this could be optimized by using `thrust::reduce_by_key()` on the ordered indices instead of `thrust::unique_copy`.  The reduction would be faster (less bandwidth) and would allocate less memory, I believe.

Reply:
I need total unique elements count. This API doesn't need unique elements, and not their individual counts. 

for `detail::get_unique_ordered_indices`, I wrote **custom `unique_copy`** which uses `thrust::copy_if` and comparators for 3 choices(keep first, last, none). drop_duplicates needed unique indices.

`thrust::reduce_by_key()` could be useful for[ `Series.value_counts`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.value_counts.html) where it needs unique elements and their counts. 

I modified and wrote using `thrust::count_if`.  it is not much faster, just few ms. I guess, most bandwidth & time is taken by row comparison.